src/main/java/com/mraof/minestuck/item/GutterThumbDriveItem.java

```diff
@@ -0,0 +1,73 @@
+package com.mraof.minestuck.item;
+
+import com.mraof.minestuck.blockentity.ComputerBlockEntity;
+import com.mraof.minestuck.player.IdentifierHandler;
+import com.mraof.minestuck.skaianet.Session;
+import com.mraof.minestuck.skaianet.SessionHandler;
+import net.minecraft.client.gui.screens.Screen;
+import net.minecraft.core.BlockPos;
+import net.minecraft.network.chat.Component;
+import net.minecraft.network.chat.TranslatableComponent;
+import net.minecraft.sounds.SoundEvents;
+import net.minecraft.sounds.SoundSource;
+import net.minecraft.world.InteractionHand;
+import net.minecraft.world.InteractionResult;
+import net.minecraft.world.InteractionResultHolder;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.item.Item;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.item.TooltipFlag;
+import net.minecraft.world.item.context.UseOnContext;
+import net.minecraft.world.level.Level;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.List;
+import java.util.function.Supplier;
+
+
+public class GutterThumbDriveItem extends Item
+{
+	public GutterThumbDriveItem(Properties pProperties)
+	{
+		super(pProperties);
+	}
+	@Override
+	public InteractionResult useOn(UseOnContext pContext)
+	{
+		if(!pContext.getLevel().isClientSide)
+		{
+			Player player = pContext.getPlayer();
+			Level level = pContext.getLevel();
+			InteractionHand pUsedHand = player.getUsedItemHand();
+			int mod = 1;
+			
+			for(BlockPos blockPos : BlockPos.betweenClosed(player.blockPosition().offset(2 * mod, mod, 2 * mod), player.blockPosition().offset(-2 * mod, -1 * mod, -2 * mod)))
```

Comment:
It should be possible to get the block that has been clicked on through pContext. The way this is currently handled, it appears that a player could consume the item accidentally if they right click on any block while near a computer

Reply:
updated